### PR TITLE
fix #2556 webview npe

### LINKED ISSUE #2556: Crash report 3.9: NPE in WPWebViewClient.shouldInterceptRequest
```
java.lang.ExceptionInInitializerError
       at android.webkit.CallbackProxy.shouldOverrideUrlLoading(CallbackProxy.java:1123)
       at android.webkit.BrowserFrame.handleUrl(BrowserFrame.java:648)
       at android.webkit.WebViewCore.nativeMouseClick(WebViewCore.java)
       at android.webkit.WebViewCore.access$7700(WebViewCore.java:63)
       at android.webkit.WebViewCore$EventHub.dispatchWebKitEvent(WebViewCore.java:1996)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvent(WebViewInputDispatcher.java:706)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvents(WebViewInputDispatcher.java:641)
       at android.webkit.WebViewInputDispatcher.access$900(WebViewInputDispatcher.java:78)
       at android.webkit.WebViewInputDispatcher$WebKitHandler.handleMessage(WebViewInputDispatcher.java:1170)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.webkit.WebViewCore$WebCoreThread.run(WebViewCore.java:930)
       at java.lang.Thread.run(Thread.java:838)
Caused by: java.lang.NullPointerException
       at org.wordpress.android.util.WPWebViewClient.shouldInterceptRequest(WPWebViewClient.java:104)
       at android.webkit.CallbackProxy.shouldInterceptRequest(CallbackProxy.java:1203)
       at android.webkit.BrowserFrame.shouldInterceptRequest(BrowserFrame.java:876)
       at android.webkit.WebViewCore.nativeMouseClick(WebViewCore.java)
       at android.webkit.WebViewCore.access$7700(WebViewCore.java:63)
       at android.webkit.WebViewCore$EventHub.dispatchWebKitEvent(WebViewCore.java:1996)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvent(WebViewInputDispatcher.java:706)
       at android.webkit.WebViewInputDispatcher.dispatchWebKitEvents(WebViewInputDispatcher.java:641)
       at android.webkit.WebViewInputDispatcher.access$900(WebViewInputDispatcher.java:78)
       at android.webkit.WebViewInputDispatcher$WebKitHandler.handleMessage(WebViewInputDispatcher.java:1170)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:194)
       at android.webkit.WebViewCore$WebCoreThread.run(WebViewCore.java:930)
       at java.lang.Thread.run(Thread.java:838)
```

### PR DESCRIPTION
**Pull request by maxme**
Use a `HttpURLConnection` instead  of `DefaultHttpClient`. That will fix #2556 issue.  

Note: the `WebResourceResponse` constructor can take 3 null parameters.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java [hunk 1]
```diff
@@ -93,22 +92,25 @@ public void onReceivedSslError(WebView view, SslErrorHandler handler, SslError e
...
-            httpGet.setHeader("Authorization", "Bearer " + mToken);
+        if (mBlog != null && mBlog.isPrivate() && !TextUtils.isEmpty(mToken) && UrlUtils.isImageUrl(stringUrl)) {
             try {
-                HttpResponse httpResponse = client.execute(httpGet);
-                InputStream responseInputStream = httpResponse.getEntity().getContent();
-                return new WebResourceResponse(httpResponse.getEntity().getContentType().toString(),
-                        "UTF-8", responseInputStream);
+                URL url = new URL(stringUrl);
+                HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+                urlConnection.setRequestProperty("Authorization", "Bearer " + mToken);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java]: When I implemented similar code [for the Reader](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderWebView.java#L216) I also set the timeout and user agent. Should that be done here as well?

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java]: Yes, I'll update the user agent, but I don't see any good reasons to change other options from the defaults.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/util/WPWebViewClient.java]: I think the timeouts default to zero (no timeout) which may [have repercussions](https://code.google.com/p/android/issues/detail?id=40874#c16)

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**